From 4691c10fb990b8b1abeb1a3761aed9a3ea28586a Mon Sep 17 00:00:00 2001
From: Hans de Goede <johannes.goede@oss.qualcomm.com>
Date: Fri, 1 May 2026 14:08:52 +0200
Subject: [PATCH 33/39] software_isp: Rename simple IPA setIspParams signal to
 paramsComputed

Rename the simple IPA setIspParams signal to paramsComputed and pass
the frame-number as argument to make the simple IPA behavior consistent
with other IPA implementations.

Signed-off-by: Hans de Goede <johannes.goede@oss.qualcomm.com>
(cherry picked from commit 77c5ac6ac9bffdbb44f19d9b1bcd79473fa62a87)
---
 include/libcamera/internal/software_isp/software_isp.h | 2 +-
 include/libcamera/ipa/soft.mojom                       | 2 +-
 src/ipa/soft/soft_simple.cpp                           | 2 +-
 src/libcamera/software_isp/software_isp.cpp            | 4 ++--
 4 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/include/libcamera/internal/software_isp/software_isp.h b/include/libcamera/internal/software_isp/software_isp.h
index 70a9175a9..90c7393f7 100644
--- a/include/libcamera/internal/software_isp/software_isp.h
+++ b/include/libcamera/internal/software_isp/software_isp.h
@@ -89,7 +89,7 @@ public:
 	Signal<const ControlList &, const ControlList &, const ControlList &> setSensorControls;
 
 private:
-	void saveIspParams();
+	void saveIspParams(uint32_t frame);
 	void setSensorCtrls(const ControlList &sensorControls,
 			    const ControlList &lensControls,
 			    const ControlList &irisControls);
diff --git a/include/libcamera/ipa/soft.mojom b/include/libcamera/ipa/soft.mojom
index 3c1e84994..bd4bc42b1 100644
--- a/include/libcamera/ipa/soft.mojom
+++ b/include/libcamera/ipa/soft.mojom
@@ -39,6 +39,6 @@ interface IPASoftEventInterface {
 	setSensorControls(libcamera.ControlList sensorControls,
 			  libcamera.ControlList lensControls,
 			  libcamera.ControlList irisControls);
-	setIspParams();
+	paramsComputed(uint32 frame);
 	metadataReady(uint32 frame, libcamera.ControlList metadata);
 };
diff --git a/src/ipa/soft/soft_simple.cpp b/src/ipa/soft/soft_simple.cpp
index 5cea85ec6..ca262be61 100644
--- a/src/ipa/soft/soft_simple.cpp
+++ b/src/ipa/soft/soft_simple.cpp
@@ -379,7 +379,7 @@ void IPASoftSimple::computeParams(const uint32_t frame)
 		algo->prepare(context_, frame, frameContext, params_);
 	params_->combinedMatrix = context_.activeState.combinedMatrix;
 
-	setIspParams.emit();
+	paramsComputed.emit(frame);
 }
 
 void IPASoftSimple::processStats(const uint32_t frame,
diff --git a/src/libcamera/software_isp/software_isp.cpp b/src/libcamera/software_isp/software_isp.cpp
index 503ff45d9..528c1930d 100644
--- a/src/libcamera/software_isp/software_isp.cpp
+++ b/src/libcamera/software_isp/software_isp.cpp
@@ -201,7 +201,7 @@ SoftwareIsp::SoftwareIsp(PipelineHandler *pipe, CameraSensor *sensor,
 		return;
 	}
 
-	ipa_->setIspParams.connect(this, &SoftwareIsp::saveIspParams);
+	ipa_->paramsComputed.connect(this, &SoftwareIsp::saveIspParams);
 	ipa_->metadataReady.connect(this,
 				    [this](uint32_t frame, const ControlList &metadata) {
 					    metadataReady.emit(frame, metadata);
@@ -462,7 +462,7 @@ void SoftwareIsp::process(uint32_t frame, FrameBuffer *input, FrameBuffer *outpu
 			       ConnectionTypeQueued, frame, input, output, debayerParams_);
 }
 
-void SoftwareIsp::saveIspParams()
+void SoftwareIsp::saveIspParams([[maybe_unused]] uint32_t frame)
 {
 	debayerParams_ = *sharedParams_;
 }
